extensions/credentials/deployment/src/main/java/io/quarkus/credentials/CredentialsProcessor.java

```diff
@@ -1,16 +1,13 @@
 package io.quarkus.credentials;
 
-import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
 import io.quarkus.deployment.annotations.BuildStep;
 
 public class CredentialsProcessor {
 
     @BuildStep
-    AdditionalBeanBuildItem registerAdditionalBeans() {
-        return new AdditionalBeanBuildItem.Builder()
-                .setUnremovable()
-                .addBeanClass(CredentialsProvider.class)
-                .build();
+    UnremovableBeanBuildItem unremoveable() {
+        return UnremovableBeanBuildItem.beanTypes(CredentialsProvider.class);
```

Comment:
I'm not sure I understand why the previous code didn't work?

Also I don't think the semantic is the same. With the previous code, we were forcing any `CredentialsProvider` to be a bean. With your code, IIUC, you have to declare the beans explicitly (and they are then made unremovable).

Reply:
tl;dr: The previous code did absolutely nothing.

Long version: The previous code was well intended, but was simple wrong. What I think it wanted to do was lookup all the implementations of `CredentialsProvider` and make them unremovable, but the way it's written, it just made the interface itself unremovable, which has zero effect.
what I have done here is what we do in all places were we expect users to provide beans but that don't explicitly declare an injection point.